app/controllers/V1/searches_controller.rb

```diff
@@ -0,0 +1,112 @@
+module V1
+  class SearchesController < ApplicationController
+    def create
+      render json: build_results(search_query), status: :created
+    end
+
+  private
+
+    def search_query
+      relation = Search.where(date_submitted: (submitted_from..submitted_to))
+      relation = relation.where(date_updated: (updated_from..updated_to))
+      relation = relation.where(submission_type:) if submission_type
+      relation = relation.where(status:) if status
+      relation = relation.where("has_been_assigned_to ? :caseworker_id", caseworker_id:) if caseworker_id
+      relation = relation.order(**sort_clause)
+
+      relation.where_terms(query)
+    end
+
+    def build_results(data)
+      {
+        metadata: {
+          total_results: data.size,
+          page: offset,
+          per_page: limit,
+        },
+        data: data.limit(limit).offset(offset),
+      }.to_json
+    end
+
+    def submission_type
+      filters[:submission_type]
+    end
+
+    def submitted_from
+      filters[:submitted_from]&.to_date
+    end
+
+    def submitted_to
+      filters[:submitted_to]&.to_date
+    end
+
+    def updated_from
+      filters[:updated_from]&.to_date
+    end
+
+    def updated_to
+      filters[:updated_to]&.to_date
+    end
+
+    def status
+      filters[:status]
+    end
+
+    def caseworker_id
+      filters[:caseworker_id]
+    end
+
+    def query
+      search_params[:query]
+    end
+
+    def sort_clause
+      return { date_updated: :desc } unless search_params[:sort_by]
+
+      { sort_by => sort_direction }
+    end
+
+    def sort_by
+      search_params.fetch(:sort_by, :date_updated)
+    end
+
+    def sort_direction
+      @sort_direction ||= search_params
+                            .fetch(:sort_direction, "asc")
+                            .downcase
+                            .gsub("ascending", "asc")
+                            .gsub("descending", "desc")
+    end
+
+    def limit
+      search_params.fetch(:per_page, 20).to_i
+    end
+
+    def offset
+      search_params.fetch(:page, 0).to_i
```

Comment:
As mentioned, you'll need to multiply your offset by `per_page`, so that for the first page of results the offset is 0, for the second it's 20, for the third it's 40 and so on.

Reply:
As mentioned in discussions sicne, the offset/page and limit/per_page are supplied by the client that is then responsible for telling the API what it wants - so i don't think this logic applies?!

Reply:
right, yes, got what you mean now. i will use page and per_page, 1 indexed, and calculate the offset.

Reply:
applied changes now